homeassistant/components/aws/__init__.py

```diff
@@ -61,6 +70,27 @@
     }
 )
 
+
+IMAGE_PROCESSING_PLATFORM_SCHEMA = IMAGE_PROCESSING_PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Required(CONF_SERVICE): vol.All(
+            cv.string, vol.Lower, vol.In(SUPPORTED_SERVICES)
+        ),
+        vol.Required(CONF_REGION): vol.All(cv.string, vol.Lower),
+        vol.Inclusive(CONF_ACCESS_KEY_ID, ATTR_CREDENTIALS): cv.string,
+        vol.Inclusive(CONF_SECRET_ACCESS_KEY, ATTR_CREDENTIALS): cv.string,
+        vol.Exclusive(CONF_PROFILE_NAME, ATTR_CREDENTIALS): cv.string,
+        vol.Exclusive(CONF_CREDENTIAL_NAME, ATTR_CREDENTIALS): cv.string,
+        vol.Required(CONF_REGION): vol.All(cv.string, vol.Lower),
+        vol.Optional(CONF_COLLECTION_ID): cv.string,
+        vol.Optional(CONF_IDENTIFY_FACES): cv.boolean,
+        vol.Optional(CONF_DETECTION_ATTRIBUTES): cv.string,
+        vol.Optional(CONF_SAVE_FILE_TIMESTAMP): cv.boolean,
+        vol.Optional(CONF_SAVE_FILE_FOLDER): cv.string,
```

Comment:
vol.Optional(CONF_SAVE_FILE_FOLDER): cv.isdir,

Reply:
Thanks, didn't know about `isdir`, might also need to be cognizant of `whitelist_external_dirs` although none of the other `image_processing` components I looked at seem to do more than basic path validation (that `pathlib` provides anyway).